### PR TITLE
Added venvstacks as a package

### PR DESCRIPTION
**Pull request by thewchan**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by thewchan** [comment]
@conda-forge/help-python @conda-forge/uv Do we know why pip check can't detect uv in the run environment?

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1089355&view=logs&j=4cabe686-70ae-553a-7fd0-310379f2cbac&t=6a4fc7c9-c31a-5115-eff9-6479d72b69ff&l=1930

**Comment by bollwyvl** [comment]
Appeasing pip check in this particularly case is... a bit awkward.

We opted to skip ~16 extra builds, as uv doesn't link against, and can
indeed install its own, python.

Feel free to weigh in here:
https://github.com/conda-forge/uv-feedstock/issues/42

**Comment by thewchan** [comment]
@bollwyvl offering a `python-uv` seems to be the pattern in other similar cases here. Would that be a huge lift?

**Comment by bollwyvl** [comment]
:shrug: guess we can see... would be an update to that feedstock, and then a PR to claim the package name, but may be feasible. `uv` releases fairly frequently, and would be rather burdensome if the patching approach that would appease `pip check` didn't work well.

**Comment by bollwyvl** [comment]
Alternate approach (fewer builds, slightly more button pushing): https://github.com/conda-forge/staged-recipes/pull/28332

File: recipes/venvstacks/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+requirements:
+  host:
+    - python {{ python_min }}
+    - pdm-backend
+    - pip
+  run:
+    - python >={{ python_min }}
+    - pip >=24.1.1
+    - pdm >=2.17.3
+    - uv >=0.2.33
```
Comment (bollwyvl) [inline recipes/venvstacks/meta.yaml:29]: `python-uv` is now available.

File: recipes/venvstacks/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+{% set name = "venvstacks" %}
+{% set version = "0.2.0" %}
+{% set python_min = "3.11" %}
```
Comment (bollwyvl) [inline recipes/venvstacks/meta.yaml:3]: Can already create a `conda_build_config.yaml` which the bot would know about.

```yaml
python_min:
  - "3.11"
```

**Comment by thewchan** [comment]
@conda-forge/staged-recipes Ready for review!

File: recipes/venvstacks/meta.yaml (hunk 2 above)
Comment (ocefpaf) [reply recipes/venvstacks/meta.yaml:3]: I kind of prefer to have that in the recipe. It is easier to have that info in a single place so we can remember to update it.

**Pull request merged**